std/stdio.d

```diff
@@ -1897,6 +1897,28 @@ $(CONSOLE
         assert(b1 == true && b2 == false);
     }
 
+    @system unittest
+    {
+        static import std.file;
+
+        auto deleteme = testFilename();
+        std.file.write(deleteme, "1\n2");
+        scope(exit) std.file.remove(deleteme);
+        int input;
+        auto f = File(deleteme);
+        f.readf("%s", &input);
+
+        import std.conv : ConvException;
+        try
+        {
+            f.readf("%s", &input);
```

Comment:
> looks like six versus half a dozen - with collectException you need to pass an int by ref etc. I'll allow this.

It would have been a one-liner (one doesn't need to pass an int by ref) and the advantage is that it is verified that the exception actually gets thrown:

```d
import std.exception, std.stdio;
static import std.file;

void main()
{
    std.file.write(std.file.deleteme, "1\n2");
    auto f = File(std.file.deleteme);
    int input;
    f.readf("%s", &input);

    // A ne-liner
    auto e = collectException(f.readf("%s", &input));
    e.msg.writeln;
    assert(e.msg == "Unexpected '\\n' when converting from type LockingTextReader to type int");
}
```

https://run.dlang.io/is/2es97o